File: pset6/readability.py
from cs50 import get_string

# Get text input
text = get_string("Text: ")
# Initialize variables
lett = sent = 0
words = 1
# For loop over every character
for i in range(len(text)):
    if text[i].isalpha():
        lett += 1
    elif text[i] in (".", "?", "!"):
        sent += 1
    elif text[i] == " ":
        words += 1
# Calculate readability w/ the Coleman-Liau formula
grade_lev = round(0.0588 * (lett / words * 100) - 0.296 * (sent / words * 100) - 15.8)
# Print grade level
if grade_lev < 1:
    print("Before Grade 1")
    exit(0)
elif grade_lev >= 16:
    print("Grade 16+")
    exit(0)
else:
    print(f"Grade {grade_lev}")
    exit(0)
exit(1)

File: pset2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>

/* Declaring functions */
bool isAlpha(char c);
bool isSent(char c);

/* Main function */
int main(void)
{
    /* Get user text input */
    string text=get_string("Text: ");
    /* Initialize and declare variables for words, letters, and sentences */
    int lett=0;
    int sent=0;
    int words=1;
    /* For loop to check each character */
    for (int i=0; i<strlen(text); i++)
    {
        if (isAlpha(text[i])==true)
        {
            lett++;
        }
        else if (isSent(text[i])==true)
        {
            sent++;
        }
        else if(text[i]==' ')
        {
            words++;
        }
    }
    /* Use the Coleman-Liau formula to calculate readability */
    float index = 0.0588 * ((float)lett/(float)words*100) - 0.296 * ((float)sent/(float)words*100) - 15.8;
    int gradeLev=round(index);

    /* Print final grade level */
    if (index<1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    else if (index>=16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    else
    {
        printf("Grade %i\n", gradeLev);
    }
    return 0;
}

/* Functions to find all the is the character is a character */
bool isAlpha(char c)
{
    if((c>64 && c<91) || (c>96 && c<123))
    {
        return true;
    }
    return false;
}

bool isSent(char c)
{
    if (c=='.'|| c=='?'||c=='!')
    {
        return true;
    }
    return false;
}